### PR TITLE
FIX/TST: recursively remove ticks

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

If we are removing ticks from all the axes in a figure, we should remove them from child axes as well.  Needed for #18900.  

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by QuLogic — APPROVED** [review]

**Review by timhoffm — COMMENTED** [review]
This is internal, so it does not matter too much. But making `remove_ax()` recursive is a bit odd. I'd put the child for loop explicitly within the ax for loop instead.

**Comment by jklymak** [comment]
@timhoffm  I'm not following.  How can it be recursive if you don't call a function?

**Comment by jklymak** [comment]
... sorry, let me clarify - child axes can have their own children, so this has to be recursive.

File: lib/matplotlib/testing/decorators.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def _checked_on_freetype_version(required_freetype_version):
 def remove_ticks_and_titles(figure):
     figure.suptitle("")
     null_formatter = ticker.NullFormatter()
-    for ax in figure.get_axes():
+    def remove_ticks(ax):
         ax.set_title("")
```
Comment (timhoffm) [inline lib/matplotlib/testing/decorators.py:115]:
```suggestion
    def remove_ticks(ax):
        """Remove ticks in *ax* and all its child Axes."""
        ax.set_title("")
```

**Review by timhoffm — COMMENTED** [review]
Oh, I see. Then let's make that clear with a docstring.

**Review by timhoffm — APPROVED** [review]
Anybody can merge after CI pass.

**Pull request merged**